collector/info_schema_innodb_metrics.go

```diff
@@ -35,6 +53,21 @@ func ScrapeInnodbMetrics(db *sql.DB, ch chan<- prometheus.Metric) error {
 		); err != nil {
 			return err
 		}
+		// Special handling of the "buffer_page_io" subsystem.
+		if subsystem == "buffer_page_io" {
+			match := bufferPageRE.FindStringSubmatch(name)
+			switch match[1] {
```

Comment:
Wouldn't you want `if len(match) != 2` to detect that the regexp failed?  It looks lke [FindStringSubmatch](https://golang.org/pkg/regexp/#Regexp.FindStringSubmatch) returns a nil if the regexp fails.  So `len(match) != 2) would be a good choice.